tests/virt_autotest/login_console.pm

```diff
@@ -35,6 +35,7 @@ sub login_to_console() {
     assert_screen([qw(grub2 grub1)], 120);
     if (!get_var("reboot_for_upgrade_step")) {
         if (get_var("XEN") || check_var("HOST_HYPERVISOR", "xen")) {
+            send_key 'up';
```

Comment:
please add a comment on top. Also, I am wondering if you are not replicating the logic from e.g. tests/installation/grub_test.pm here too much and should rather use more of the same code